### PR TITLE
Improvements for date time fields

### LINKED ISSUE #776: Datetime default value now errors
**Describe the bug**

The CLI "xata schema edit" errors when the datetime default value is set to now:
image

### LINKED ISSUE #812: Type errors for datetime values
**Description**

The client has type warnings when specifying records that have datetime values.

This is from the playground:

image

With double (or single) quotes:
day: "2022-12-12T00:00:00.000Z" -> `type string not assignable to type Date`

Without quotes:
day: 2022-12-12T00:00:00.000Z -> `type number not assignable to type Date`

### PR DESCRIPTION
**Pull request by SferaDev**
- Add tsc check for 4.8
- Fix types in TS 4.7 for sort
- Fix `now` modifier for default value in CLI
- Allow strings in datefield types

Closes #776 
Closes #812

### CODE REVIEW

File: .github/workflows/tsc.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        target: ['next', 'beta', 'latest', '~4.7']
+        target: ['next', 'beta', 'latest', '~4.8', '~4.7']
```
Comment (SferaDev) [inline .github/workflows/tsc.yml:16]: Now that 4.9 has been released, adding 4.8. Whenever a new version is bumped we need to add the previous one to the matrix.

File: cli/src/commands/schema/edit.ts [hunk 2]
```diff
@@ -750,6 +750,9 @@ function parseDefaultValue(type: string, val: string): string | undefined {
   } else if (type === 'link') {
     return val ? String(val) : undefined;
   } else if (type === 'datetime') {
+    // Date fields have special values
+    if (['now'].includes(val)) return val;
```
Comment (SferaDev) [inline cli/src/commands/schema/edit.ts:754]: Adding a fast exit if one of the aliases are detected (like now which is not a date), leaving the door open to add more in the future.

File: packages/client/src/schema/query.ts [hunk 3]
```diff
@@ -79,7 +78,7 @@ export class Query<Record extends XataRecord, Result extends XataRecord = Record
     this.#data.filter.$all = data.filter?.$all ?? parent?.filter?.$all;
     this.#data.filter.$not = data.filter?.$not ?? parent?.filter?.$not;
     this.#data.filter.$none = data.filter?.$none ?? parent?.filter?.$none;
-    this.#data.sort = data.sort ?? parent?.sort;
+    this.#data.sort = (data.sort ?? parent?.sort) as QueryOptions<Record>['sort']; // Fix for TS =4.7
```
Comment (SferaDev) [inline packages/client/src/schema/query.ts:81]: This is a better fix than link 33

File: packages/client/src/schema/record.ts [hunk 4]
```diff
@@ -128,15 +128,21 @@ export function isXataRecord(x: any): x is XataRecord & Record<string, unknown>
...
 }
 
+type EditableDataFields<T> = T extends XataRecord
+  ? { id: string } | string
+  : NonNullable<T> extends XataRecord
+  ? { id: string } | string | null | undefined
+  : T extends Date
+  ? string | Date
+  : NonNullable<T> extends Date
+  ? string | Date | null | undefined
```
Comment (SferaDev) [inline packages/client/src/schema/record.ts:138]: Allow string dates

File: test/integration/dates.test.ts [hunk 5]
```diff
@@ -36,6 +36,14 @@ describe('dates', () => {
     expect(record.birthDate?.toISOString()).toEqual(birthDate.toISOString());
   });
 
+  test("add a record with a string date (it's parsed)", async () => {
+    const birthDate = new Date();
+    const record = await xata.db.users.create({ full_name: 'foo2', birthDate: birthDate.toISOString() });
+
+    expect(record.birthDate instanceof Date).toEqual(true);
+    expect(record.birthDate?.toISOString()).toEqual(birthDate.toISOString());
+  });
```
Comment (SferaDev) [inline test/integration/dates.test.ts:45]: A test with a string date

**Review by richardgill — APPROVED** [review]
LGTM nice PR appreciate the comments

**Pull request merged**